x-pack/platform/plugins/private/gen_ai_settings/public/components/default_at_connector/default_ai_connector.tsx

```diff
@@ -0,0 +1,238 @@
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  GEN_AI_SETTINGS_DEFAULT_AI_CONNECTOR,
+  GEN_AI_SETTINGS_DEFAULT_AI_CONNECTOR_DEFAULT_ONLY,
+} from '@kbn/management-settings-ids';
+import { isEqual } from 'lodash';
+import type { FieldDefinition, UnsavedFieldChange } from '@kbn/management-settings-types';
+import type { UiSettingsType } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import type { UseGenAiConnectorsResult } from '../../hooks/use_genai_connectors';
+import { useSettingsContext } from '../../contexts/settings_context';
+import { NO_DEFAULT_CONNECTOR } from '../../../common/constants';
+import { useKibana } from '../../hooks/use_kibana';
+
+interface Props {
+  connectors: UseGenAiConnectorsResult;
+}
+
+const NoDefaultOption: EuiComboBoxOptionOption<string> = {
+  label: i18n.translate(
+    'xpack.gen_ai_settings.settings.defaultLLm.select.option.noDefaultConnector',
+    { defaultMessage: 'No default connector' }
+  ),
+  value: NO_DEFAULT_CONNECTOR,
+};
+
+const getOptions = (connectors: UseGenAiConnectorsResult): EuiComboBoxOptionOption<string>[] => {
+  const preconfigured =
+    connectors.connectors
+      ?.filter((connector) => connector.isPreconfigured)
+      .map((connector) => ({
+        label: connector.name,
+        value: connector.id,
+      })) ?? [];
+
+  const custom =
+    connectors.connectors
+      ?.filter((connector) => !connector.isPreconfigured)
+      .map((connector) => ({
+        label: connector.name,
+        value: connector.id,
+      })) ?? [];
+
+  return [
+    NoDefaultOption,
+    {
+      label: i18n.translate(
+        'xpack.gen_ai_settings.settings.defaultLLm.select.group.preconfigured.label',
+        { defaultMessage: 'Pre-configured' }
+      ),
+      value: 'preconfigured',
+      options: preconfigured,
+    },
+    {
+      label: i18n.translate('xpack.gen_ai_settings.settings.defaultLLm.select.group.custom.label', {
+        defaultMessage: 'Custom connectors',
+      }),
+      value: 'custom',
+      options: custom,
+    },
+  ];
+};
+
+const getOptionsByValues = (
+  value: string,
+  options: EuiComboBoxOptionOption<string>[]
+): EuiComboBoxOptionOption<string>[] => {
+  const getOptionsByValuesHelper = (
+    option: EuiComboBoxOptionOption<string>
+  ): EuiComboBoxOptionOption<string>[] => {
+    if (option.options === undefined && option.value === value) {
+      // If the option has no sub-options and its value is in the selected values, include it
+      return [option];
+    }
+    if (option.options) {
+      // If the option has sub-options, recursively get their options
+      return option.options.flatMap(getOptionsByValuesHelper);
+    }
+    return [];
+  };
+
+  return options.flatMap(getOptionsByValuesHelper);
+};
+
+export const DefaultAIConnector: React.FC<Props> = ({ connectors }) => {
+  const [options, setOptions] = useState<EuiComboBoxOptionOption<string>[]>(getOptions(connectors));
```

Comment:
Maybe we can get rid of this useState + useEffect as options are derived purely from `connectors`. 

```suggestion
const options = useMemo(() => getOptions(connectors), [connectors])
```

Reply:
fixed in https://github.com/elastic/kibana/pull/231940/commits/98dd5efc3d5d7345756d219632d56a73c9a588fd